Source/Parser/Functions/IterableJoiningFunction.cs

```diff
@@ -59,7 +59,10 @@ public override bool Evaluate(InterpreterScope scope, out ExpressionBase result)
                 iteratorScope.AssignVariable(predicateParameter, result);
 
                 if (!predicate.Evaluate(iteratorScope, out result))
+                {
+                    result = new ErrorExpression(string.Format("predicate function did not return any value for the predicate input {0}", input.ToString()));
```

Comment:
The `ToString()` on `input` is redundant. `String.Format` will implicitly call the `ToString()` method on each non-string parameter.